api/mfx_dispatch/windows/src/mfx_dispatcher.cpp

```diff
@@ -318,3 +325,292 @@ mfxStatus MFX_DISP_HANDLE::UnLoadSelectedDLL(void)
     return mfxRes;
 
 } // mfxStatus MFX_DISP_HANDLE::UnLoadSelectedDLL(void)
+
+#if (defined(_WIN64) || defined(_WIN32)) && (MFX_VERSION >= MFX_VERSION_NEXT)
+static mfxStatus InitDummySession(mfxU32 adapter_n, MFXVideoSession & dummy_session)
```

Comment:
errgghh.... I do NOT like creating any dummy sessions in the dispatcher for the reason that they are not cheap. They create scheduler and fully qualified core. If that's needed to support adapter query via bitstream I would better sacrifice this function for now.

Reply:
To solve that I would suggest to implement MFXQueryAdapters* functions directly inside mediasdk and teach dispatcher to collect this info directly from msdk libs and return aggregated info.

Reply:
Dummy sessions used in each function. No difference for bitstream in this

Reply:
There will be difference for Linux dispatcher which is tiny and I don't want to make it bigger.

Reply:
About implementing it in lib. How do you suppose to retrieve information from adapter? This information becomes available at init time. How should we redesign MSDK to connect unitialized library to adapter? Currently dispatcher loads lib during first init call (that's why there are dummy sessions and inits here), but how to bypass them, I don't see an easy way now, which will be better that proposed solution